client/driver/executor/executor.go

```diff
@@ -0,0 +1,221 @@
+}
+
+// LaunchCmd launches a process and returns it's state. It also configures an
+// applies isolation on certain platforms.
+func (e *UniversalExecutor) LaunchCmd(command *ExecCommand, ctx *ExecutorContext) (*ProcessState, error) {
+	e.logger.Printf("[DEBUG] executor: launching command %v %v", command.Cmd, strings.Join(command.Args, ""))
+
+	e.ctx = ctx
+
+	// configuring the task dir
+	if err := e.configureTaskDir(); err != nil {
+		return nil, err
+	}
+
+	// configuring the chroot
+	if err := e.configureIsolation(); err != nil {
+		return nil, err
+	}
+
+	// setting the user of the process
+	if e.ctx.UnprivilegedUser {
+		if err := e.runAs("nobody"); err != nil {
+			return nil, err
+		}
+	}
+
+	// configuring log rotate
+	stdoPath := filepath.Join(e.taskDir, allocdir.TaskLocal, fmt.Sprintf("%v.stdout", ctx.TaskName))
+	stdo, err := os.OpenFile(stdoPath, os.O_CREATE|os.O_RDWR|os.O_APPEND, 0666)
+	if err != nil {
+		return nil, err
+	}
+	e.cmd.Stdout = stdo
+
+	stdePath := filepath.Join(e.taskDir, allocdir.TaskLocal, fmt.Sprintf("%v.stderr", ctx.TaskName))
+	stde, err := os.OpenFile(stdePath, os.O_CREATE|os.O_RDWR|os.O_APPEND, 0666)
+	if err != nil {
+		return nil, err
+	}
+	e.cmd.Stderr = stde
+
+	// setting the env, path and args for the command
+	e.ctx.TaskEnv.Build()
+	e.cmd.Env = ctx.TaskEnv.EnvList()
+	e.cmd.Path = ctx.TaskEnv.ReplaceEnv(command.Cmd)
+	e.cmd.Args = append([]string{e.cmd.Path}, ctx.TaskEnv.ParseAndReplace(command.Args)...)
+	if filepath.Base(command.Cmd) == command.Cmd {
+		if lp, err := exec.LookPath(command.Cmd); err != nil {
+		} else {
+			e.cmd.Path = lp
+		}
+	}
+
+	// starting the process
+	if err := e.cmd.Start(); err != nil {
+		return nil, fmt.Errorf("error starting command: %v", err)
+	}
+
+	go e.wait()
+	return &ProcessState{Pid: e.cmd.Process.Pid, ExitCode: -1, Time: time.Now()}, nil
+}
+
+// Wait waits until a process has exited and returns it's exitcode and errors
+func (e *UniversalExecutor) Wait() (*ProcessState, error) {
+	<-e.processExited
+	return e.exitState, nil
+}
+
+func (e *UniversalExecutor) wait() {
+	defer close(e.processExited)
+	err := e.cmd.Wait()
+	if err == nil {
+		e.exitState = &ProcessState{Pid: 0, ExitCode: 0, Time: time.Now()}
+		return
+	}
+	exitCode := 1
+	if exitErr, ok := err.(*exec.ExitError); ok {
+		if status, ok := exitErr.Sys().(syscall.WaitStatus); ok {
+			exitCode = status.ExitStatus()
+		}
+	}
+	if e.ctx.FSIsolation {
+		e.removeChrootMounts()
+	}
+	if e.ctx.ResourceLimits {
+		e.destroyCgroup()
+	}
+	e.exitState = &ProcessState{Pid: 0, ExitCode: exitCode, Time: time.Now()}
+}
+
+// Exit cleans up the alloc directory, destroys cgroups and kills the user
+// process
+func (e *UniversalExecutor) Exit() error {
+	var merr multierror.Error
+	if e.cmd.Process != nil {
+		proc, err := os.FindProcess(e.cmd.Process.Pid)
+		if err != nil {
+			e.logger.Printf("[ERROR] can't find process with pid: %v, err: %v", e.cmd.Process.Pid, err)
```

Comment:
@dadgar I was thinking about this. But the logs from the executor isn't going to be mixed with logs of other components, so I didn't add executor to it.